utils/planets/earth.py

```diff
@@ -0,0 +1,43 @@
+
+import firedrake as fd
+
+import firedrake_utils.units as units
+
+# # # === --- constants --- === # # #
+
+# length of a single earth day
+day = 24*units.hour
+Day = fd.Constant(day)
+
+# radius in metres
+radius = 6371220*units.metre
+Radius = fd.Constant(radius)
+
+# rotation rate
+omega = 7.292e-5/units.second
+Omega = fd.Constant(omega)
+
+# gravitational acceleration
+gravity = 9.80616*units.metre/(units.second*units.second)
+Gravity = fd.Constant(gravity)
+
+# # # === --- planetary mesh --- === # # #
+
+
+def IcosahedralMesh(refinement_level=0,
+                    degree=1,
+                    reorder=None,
+                    distribution_parameters=None,
+                    comm=fd.COMM_WORLD):
+
+    globe = \
+        fd.IcosahedralSphereMesh(radius=radius,
```

Comment:
This needs a rethink because in mg you are also creating sphere meshes, so if we are going to use it then it needs to allow for the mesh hierarchy as well.

Reply:
Ah, this one is left over from my own utils repo. I'll remove it so the one in mg is the only sphere mesh generator